extensions/b2g/viewer.js

```diff
@@ -0,0 +1,213 @@
+  pdfViewer: null,
+  pdfHistory: null,
+  pdfLinkService: null,
+  loading: true,
+  updateScaleControls: false,
+
+  open: function (params) {
+    var url = params.url, originalUrl = params.originalUrl;
+
+    this.initUI();
+    this.setTitleUsingUrl(originalUrl);
+
+    // Loading document.
+    PDFJS.getDocument(url).then(function (pdfDocument) {
+      // Document loaded, specifying document for the viewer.
+      this.pdfDocument = pdfDocument;
+      this.pdfViewer.setDocument(pdfDocument);
+      this.pdfLinkService.setDocument(pdfDocument);
+      this.pdfHistory.initialize(pdfDocument.fingerprint);
+    }.bind(this));
+  },
+
+  setTitleUsingUrl: function pdfViewSetTitleUsingUrl(url) {
+    this.url = url;
+    var title = PDFJS.getFileName(url) || url;
+    try {
+      title = decodeURIComponent(title);
+    } catch (e) {
+      // decodeURIComponent may throw URIError,
+      // fall back to using the unprocessed url in that case
+    }
+    this.setTitle(title);
+  },
+
+  setTitle: function pdfViewSetTitle(title) {
+    document.title = title;
+    document.getElementById('activityTitle').textContent = title;
+  },
+
+  get pagesCount() {
+    return this.pdfDocument.numPages;
+  },
+
+  set page(val) {
+    this.pdfViewer.currentPageNumber = val;
+  },
+
+  get page() {
+    return this.pdfViewer.currentPageNumber;
+  },
+
+  zoomIn: function pdfViewZoomIn(ticks) {
+    var newScale = this.pdfViewer.currentScale;
+    do {
+      newScale = (newScale * DEFAULT_SCALE_DELTA).toFixed(2);
+      newScale = Math.ceil(newScale * 10) / 10;
+      newScale = Math.min(MAX_SCALE, newScale);
+    } while (--ticks && newScale < MAX_SCALE);
+    this.setScale(newScale, true);
+  },
+
+  zoomOut: function pdfViewZoomOut(ticks) {
+    var newScale = this.pdfViewer.currentScale;
+    do {
+      newScale = (newScale / DEFAULT_SCALE_DELTA).toFixed(2);
+      newScale = Math.floor(newScale * 10) / 10;
+      newScale = Math.max(MIN_SCALE, newScale);
+    } while (--ticks && newScale > MIN_SCALE);
+    this.setScale(newScale, true);
+  },
+
+  setScale: function (value, resetAutoSettings) {
+    this.updateScaleControls = !!resetAutoSettings;
+    this.pdfViewer.currentScaleValue = value;
+    this.updateScaleControls = true;
+  },
+
+  initUI: function pdfViewInitUI() {
+    var linkService = new PDFJS.PDFLinkService();
+    this.pdfLinkService = linkService;
+
+    var container = document.getElementById('viewerContainer');
+    var pdfViewer = new PDFJS.PDFViewer({
+      container: container,
+      linkService: linkService
+    });
+    this.pdfViewer = pdfViewer;
+    linkService.setViewer(pdfViewer);
+
+    this.pdfHistory = new PDFJS.PDFHistory({
+      linkService: linkService
+    });
+    linkService.setHistory(this.pdfHistory);
+
+    container.addEventListener('pagesinit', function () {
+      // we can use pdfViewer now, e.g. let's change default scale.
+      pdfViewer.currentScaleValue = 'page-width';
+    });
```

Comment:
Just asking: did you intended to change the initial zoom level from `auto` to `page-width`, or is it a result of copy-pasting from the [simpleviewer component](https://github.com/mozilla/pdf.js/blob/master/examples/components/simpleviewer.js)?